src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMutation.cs

```diff
@@ -229,6 +232,111 @@ public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[
         }
     }
 
+    internal class CollectImpl : CollectFunction, IAsyncTexlFunction3
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            return await new CollectProcess().Process(irContext, args, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    internal class CollectScalarImpl : CollectScalarFunction, IAsyncTexlFunction3
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            return await new CollectProcess().Process(irContext, args, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    internal class CollectProcess
+    {
+        internal async Task<FormulaValue> Process(FormulaType irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            FormulaValue arg0;
+            var argc = args.Length;
+
+            // Need to check if the Lazy first argument has been evaluated since it may have already been
+            // evaluated in the ClearCollect case.
+            if (args[0] is LambdaFormulaValue arg0lazy)
+            {
+                arg0 = await arg0lazy.EvalAsync().ConfigureAwait(false);
+            }
+            else
+            {
+                arg0 = args[0];
+            }
+
+            if (arg0 is BlankValue)
+            {
+                return arg0;
+            }
+
+            if (arg0 is ErrorValue)
+            {
+                return arg0;
+            }
+
+            if (arg0 is not TableValue)
+            {
+                return CommonErrors.RuntimeTypeMismatch(IRContext.NotInSource(arg0.Type));
+            }
+
+            var tableValue = arg0 as TableValue;
+
+            List<DValue<RecordValue>> resultRows = new List<DValue<RecordValue>>();
```

Comment:
> List<DValue<RecordValue>> resultRows [](http://example.com/codeflow?start=11&length=37)

interpreter only has v1 semantics. 
Collect() always returns single-record in v1.